CMake/AbseilHelpers.cmake

```diff
@@ -0,0 +1,147 @@
+#
+# Copyright 2017 The Abseil Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include(CMakeParseArguments)
+
+set(_ABSEIL_HELPERS_PATH "${CMAKE_CURRENT_LIST_DIR}")
+
+#
+# create a static library based on the following variable
+#
+# parameters
+# SOURCES : sources files for the library
+# PUBLIC_LIBRARIES: targets and flags for linking phase
+# PRIVATE_COMPILE_FLAGS: compile flags for the library. Will not be exported.
+# DISABLE_INSTALL: if set, disable the installation of the library
+#
+# create a target associated to <NAME>
+# libraries are installed under CMAKE_INSTALL_FULL_LIBDIR by default
+#
+function(abseil_library)
+    cmake_parse_arguments(ABSL_LIB
+        "DISABLE_INSTALL"
+        "TARGET"
+        "SOURCES;PUBLIC_LIBRARIES;PRIVATE_COMPILE_FLAGS"
+        ${ARGN}
+    )
+
+    set(_NAME ${ABSL_LIB_TARGET})
+    string(TOUPPER ${_NAME} _UPPER_NAME)
+
+    add_library(${_NAME} STATIC ${ABSL_LIB_SOURCES})
+
+    target_compile_options(${_NAME} PRIVATE ${ABSL_COMPILE_CXXFLAGS} ${ABSL_LIB_PRIVATE_COMPILE_FLAGS} )
+    target_link_libraries(${_NAME} ${ABSL_LIB_PUBLIC_LIBRARIES})
+
+
+
+    if(${ABSL_LIB_DISABLE_INSTALL})
+        # reserved for internal dep
+    else()
+        install(TARGETS ${_NAME}
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
```

Comment:
I guess the use of "FULL DIR" is once again because of Nix?
Even though I'm pretty sure the CMake best practices is to specify relative paths.

Reply:
>  Even though I'm pretty sure the CMake best practices is to specify relative paths.
Cmake work internally with absoute path anyway, this should not matter